File: java/src/main/java/leetcode/leet4.java
package leetcode;

import java.util.ArrayList;
import java.util.Collections;

public class leet4 {
    public static void main(String[] args) {
        int[] num1 ={1,2};
        int[] num2 ={3,4};
        System.out.println(findMedianSortedArrays(num1,num2));
    }

    public static double findMedianSortedArrays(int[] nums1, int[] nums2) {
        if(nums1.length==0&&nums2.length==0){
            return 0;
        }
        double num;
        ArrayList arrayList1=new ArrayList();
        ArrayList<Integer>  arrayList2;
        arrayList2 = new ArrayList<>();
        for (int j : nums1) {
            arrayList1.add(j);
        }
        for (int j : nums2) {
            arrayList2.add(j);
        }
        arrayList1.addAll(arrayList2);
        Collections.sort(arrayList1);
        int size=arrayList1.size();
        if(size%2==0){
            int x=size/2;
            int y=x+1;
            num=((double) arrayList1.get(x-1)+(double)arrayList1.get(y-1))/2;
        }
        else{
            int x=(size+1)/2;
            num=(double) arrayList1.get(x-1);
        }

        return num;
    }
}

File: python/python/leetcode/leet4.py
import math
from typing import List


def find_median_sorted_arrays(self, nums1: List[int], nums2: List[int]) -> float:
    nums1 = nums1 + nums2
    nums1.sort()
    x = math.ceil(len(nums1) / 2) - 1
    if len(nums1) % 2 == 0:
        return (nums1[x] + nums1[x + 1]) / 2
    else:
        return nums1[x]


# 给定两个大小分别为 m 和 n 的正序（从小到大）数组nums1 和nums2。请你找出并返回这两个正序数组的 中位数
# 算法的时间复杂度应该为 O(log (m+n))
if __name__ == '__main__':
    nums1 = [1, 3]
    nums2 = [2]
    print(find_median_sorted_arrays(0, nums1, nums2))
